src/npctalk.cpp

```diff
@@ -2912,6 +2930,89 @@ void talk_function::stop_guard( npc &p )
     p.guard_pos = npc::no_goal_point;
 }
 
+void talk_function::become_overseer( npc &p )
+{
+    if( !g->u.has_trait( trait_FACTION_CAMP_TUTORIAL ) ){
+        g->u.set_mutation( trait_FACTION_CAMP_TUTORIAL );
+        if( query_yn( _("Would you like to review the faction camp description?") ) ){
+            faction_camp_tutorial();
+        }
+    }
+
+    const point omt_pos = ms_to_omt_copy( g->m.getabs( p.posx(), p.posy() ) );
+    oter_id &omt_ref = overmap_buffer.ter( omt_pos.x, omt_pos.y, p.posz() );
+
+    if( omt_ref.id() != "field" ){
+        popup( _("You must build your camp in an empty field.") );
+        return;
+    }
+
+    std::vector<std::pair<std::string, tripoint>> om_region = om_building_region( p, 1 );
+    for( const auto &om_near : om_region ){
+        if ( om_near.first != "field" && om_near.first != "forest" && om_near.first != "forest_thick" &&
+                om_near.first != "forest_water" && om_near.first.find("river_") == std::string::npos ){
+            popup( _("You need more room for camp expansions!") );
+            return;
+        }
+    }
+    std::vector<std::pair<std::string, tripoint>> om_region_extended = om_building_region( p, 3 );
+    int forests = 0;
+    int waters = 0;
+    int swamps = 0;
+    int fields = 0;
+    for( const auto &om_near : om_region_extended ){
+        if( om_near.first.find("faction_base_camp") != std::string::npos ){
+            popup( _("You are too close to another camp!") );
+            return;
+        }
+        if( om_near.first == "forest" || om_near.first == "forest_thick" ){
+            forests++;
+        } else if( om_near.first.find("river_") != std::string::npos ){
+            waters++;
+        } else if( om_near.first == "forest_water" ){
+            swamps++;
+        } else if( om_near.first == "field" ){
+            fields++;
+        }
+    }
+
+    bool display = false;
+    std::string buffer = _("Warning, you have selected a region with the following issues:\n \n");
+    if( forests < 3 ){
+        display = true;
+        buffer = buffer + _("There are few forests.  Wood is your primary construction material.\n");
```

Comment:
Maybe we should color these messages to show which is bad?

And also make some pivot table showing location stats?

```
Forests: <good>enough</good>
Plains: <good>enough</good>
Rivers: <bad>low</bad>
Swamps: <critical>none</critical>
```